inc/hclib_cpp.h

```diff
@@ -31,6 +31,38 @@ int current_worker();
 int num_workers();
 int get_num_places(place_type_t type);
 void get_places(place_t **pls, place_type_t type);
+place_t *get_current_place();
+place_t **get_children_of_place(place_t *pl, int *num_children);
+place_t *get_root_place();
+place_t **get_nvgpu_places(int *n_nvgpu_places);
+char *get_place_name(place_t *pl);
+
+#ifdef HC_CUDA
+template<typename T>
+T *allocate_at(place_t *pl, size_t nitems, int flags) {
```

Comment:
In Habanero-C/hclib, the data driven task/future (blocking constructs) syntax was "DDF_FREE", "DDF_CREATE", DDF_HANDLE", "HC_MALLOC", "HC_FREE", etc. If we follow the same naming convention then we should change all hcpp's synchronous (blocking constructs) as "ALLOC_AT", "DEALLOC_AT" (or FREE_AT), MEMSET_AT.. etc.... ?